docs/ingressdns.md

```diff
@@ -0,0 +1,226 @@
+# Federated Ingress DNS Guide
+
+Federated Ingress DNS (FIDNS) provides the ability to programmatically manage DNS records of Ingress objects.
+FIDNS is not meant to replace in-cluster DNS providers such as [CoreDNS](https://coredns.io/). Instead, FIDNS works with
+systems such as [ExternalDNS](https://github.com/kubernetes-incubator/external-dns) that are responsible for
+managing external DNS records of Ingress objects.
+
+## Concepts
+
+<p align="center"><img src="./images/ingressdns.png" width="711"></p>
+
+The above diagram illustrates FIDNS. A typical FIDNS workflow consists of:
+
+1. Creating `FederatedDeployment`, `FederatedService` and `FederatedIngress` objects. The Federation sync
+controller propagates the corresponding objects to target clusters.
+2. Creating a `MultiClusterIngressDNSRecord` object that identifies the intended domain name and other optional resource
+record details of the `FederatedIngress`.
+3. A `DNSEndpoint` object that corresponds to the `MultiClusterIngressDNSRecord` is created by the DNS Endpoint
+Controller.
+3. An external DNS system (i.e. ExternalDNS) watches and lists `DNSEndpoint` objects and creates DNS resource records
+in supported DNS providers based on the desired state of the object.
+
+FIDNS is comprised of multiple types and controllers. The Ingress DNS Controller will watch and list
+`MultiClusterIngressDNSRecord` objects and update the status of the object with the corresponding Ingress load-balancer
+IP address for each cluster listed in the `FederatedIngressPlacement`. The DNS Endpoint Controller watches and lists the
+`MultiClusterIngressDNSRecord` object and creates a `DNSEndpoint` object for each `MultiClusterIngressDNSRecord`. An
+external DNS system (i.e. [ExternalDNS](https://github.com/kubernetes-incubator/external-dns)) is responsible for
+watching `DNSEndpoint` objects and using the endpoint information to create the DNS resource record in
+the DNS provider.
+
+## User Guide
+
+Follow the steps in the Federation-v2 [User Guide](./userguide.md) to setup your Kubernetes clusters and federation
+control-plane.
+
+ExternalDNS supports several different
+[providers](https://github.com/kubernetes-incubator/external-dns#the-latest-release-v05). If needed, create a domain
+name with one of the supported providers or delegate a DNS subdomain to use for ExternalDNS. Reference your DNS
+provider documentation on how to create a domain or delegate a subdomain. The `example` subdomain of domain name
+`example.com` is used by this guide.
+
+Deploy the ExternalDNS controller. __NOTE:__ [Google Cloud DNS](https://cloud.google.com/dns) is used as the provider
+and `external.example.com` is used as the `domain-filter`. The filter will make ExternalDNS see only the hosted zones
+matching `external.example.com`. Omit this flag to process all available hosted zones.
+```bash
+cat <<EOF | kubectl create -f -
```

Comment:
@shashidharatd I reviewed the fed-v3 helm [readme](https://github.com/kubernetes-sigs/federation-v2/blob/master/charts/federation-v2/README.md) and I don't see where/how it deploys external-dns or a sample deployment/service/ingress/multiclusteringressdnsrecord. It would also be nice to simplify the deps/steps for the time being by not using Helm and get user feedback first.